server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1685,10 +1685,8 @@ public ClusterState execute(ClusterState currentState) {
                         reusedExistingDelete = true;
                         return currentState;
                     }
-                    final List<SnapshotId> toDelete = List.copyOf(snapshotIdsRequiringCleanup);
-                    ensureBelowConcurrencyLimit(repoName, toDelete.get(0).getName(), snapshots, deletionsInProgress);
```

Comment:
Just dropping the check here. I left the deletes to still count towards the limit so a running delete may still block a snapshot or clone operation. I think since deletes are the most memory intensive of the repo write operations this is reasonable but probably doesn't really matter one way or another.